tests/compile-fail/lifetimes.rs

```diff
@@ -0,0 +1,62 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(needless_lifetimes)]
+
+fn distinct_lifetimes<'a, 'b>(_x: &'a u8, _y: &'b u8, _z: u8) { } //~ERROR
+
+fn distinct_and_static<'a, 'b>(_x: &'a u8, _y: &'b u8, _z: &'static u8) { } //~ERROR
+
+fn same_lifetime_on_input<'a>(_x: &'a u8, _y: &'a u8) { } // no error, same lifetime on two params
+
+fn only_static_on_input(_x: &u8, _y: &u8, _z: &'static u8) { } // no error, static involved
+
+fn in_and_out<'a>(x: &'a u8, _y: u8) -> &'a u8 { x } //~ERROR
+
+fn multiple_in_and_out_1<'a>(x: &'a u8, _y: &'a u8) -> &'a u8 { x } // no error, multiple input refs
+
+fn multiple_in_and_out_2<'a, 'b>(x: &'a u8, _y: &'b u8) -> &'a u8 { x } // no error, multiple input refs
+
+fn in_static_and_out<'a>(x: &'a u8, _y: &'static u8) -> &'a u8 { x } // no error, static involved
+
+fn deep_reference_1<'a, 'b>(x: &'a u8, _y: &'b u8) -> Result<&'a u8, ()> { Ok(x) } // no error
```

Comment:
Good point, I ignored these so far.

Really makes me wonder if there is a function somewhere in `librustc_*` that I could use instead :)